packages/ruby/lib/readme/payload.rb

```diff
@@ -1,15 +1,26 @@
+require "uuid"
+
 module Readme
   class Payload
-    def initialize(har, user_info, development:)
+    attr_reader :ignore
+
+    def initialize(har, info, development:)
       @har = har
       # swap api_key for id
-      user_info[:id] = user_info.delete :api_key unless user_info[:api_key].nil?
-      @user_info = user_info
+      info[:id] = info.delete :api_key unless info[:api_key].nil?
+      # pull log_id and ignore fields out of info to construct a user_info hash that can be assigned to the group key
+      @log_id = info[:log_id]
+      @ignore = info[:ignore]
+      info.delete :log_id
+      info.delete :ignore
+      @user_info = info
```

Comment:
Maybe prefer not to modify the args?

```suggestion
      # pull log_id and ignore fields out of info to construct a user_info hash that can be assigned to the group key
      @log_id = info[:log_id]
      @ignore = info[:ignore]
      @user_info = info.except(:api_key, :log_id, :ignore)
      @user_info[:id] = info[:api_key] unless info[:api_key].nil?
```

Reply:
yeah good call, thanks